libgalois/include/galois/analytics/connected_components/connected_components.h

```diff
@@ -0,0 +1,156 @@
+  // consistent way to configure.
+private:
+  Algorithm algorithm_;
+  ptrdiff_t edge_tile_size_;
+  uint32_t neighbor_sample_size_;
+  uint32_t component_sample_frequency_;
+
+  ConnectedComponentsPlan(
+      Architecture architecture, Algorithm algorithm, ptrdiff_t edge_tile_size,
+      uint32_t neighbor_sample, uint32_t component_sample_frequency)
+      : Plan(architecture),
+        algorithm_(algorithm),
+        edge_tile_size_(edge_tile_size),
+        neighbor_sample_size_(neighbor_sample),
+        component_sample_frequency_(component_sample_frequency) {}
+
+public:
+  // kChunkSize is a fixed const int (default value: 1)
+  static const int kChunkSize;
+
+  ConnectedComponentsPlan()
+      : ConnectedComponentsPlan{kCPU, kAutomatic, 0, 0, 0} {}
+
+  Algorithm algorithm() const { return algorithm_; }
+  ptrdiff_t edge_tile_size() const { return edge_tile_size_; }
+  uint32_t neighbor_sample_size() const { return neighbor_sample_size_; }
+  uint32_t component_sample_frequency() const {
+    return component_sample_frequency_;
+  }
+
+  static ConnectedComponentsPlan Serial() { return {kCPU, kSerial, 0, 0, 0}; }
+
+  static ConnectedComponentsPlan LabelProp() {
+    return {kCPU, kLabelProp, 0, 0, 0};
+  }
+
+  static ConnectedComponentsPlan Synchronous() {
+    return {kCPU, kSynchronous, 0, 0, 0};
+  }
+
+  static ConnectedComponentsPlan Asynchronous() {
+    return {kCPU, kAsynchronous, 0, 0, 0};
+  }
+
+  static ConnectedComponentsPlan EdgeAsynchronous() {
+    return {kCPU, kEdgeAsynchronous, 0, 0, 0};
+  }
+
+  static ConnectedComponentsPlan EdgeTiledAsynchronous(
+      ptrdiff_t edge_tile_size = 512) {
+    return {kCPU, kEdgeTiledAsynchronous, edge_tile_size, 0, 0};
+  }
+
+  static ConnectedComponentsPlan BlockedAsynchronous() {
+    return {kCPU, kBlockedAsynchronous, 0, 0, 0};
+  }
+
+  static ConnectedComponentsPlan Afforest(
+      uint32_t neighbor_sample_size = 2,
+      uint32_t component_sample_frequency = 1024) {
+    return {
+        kCPU, kAfforest, 0, neighbor_sample_size, component_sample_frequency};
+  }
+
+  static ConnectedComponentsPlan EdgeAfforest(
+      uint32_t neighbor_sample_size = 2,
+      uint32_t component_sample_frequency = 1024) {
+    return {
+        kCPU, kEdgeAfforest, 0, neighbor_sample_size,
+        component_sample_frequency};
+  }
+
+  static ConnectedComponentsPlan EdgeTiledAfforest(
+      ptrdiff_t edge_tile_size = 512, uint32_t neighbor_sample_size = 2,
+      uint32_t component_sample_frequency = 1024) {
+    return {
+        kCPU, kEdgeAfforest, edge_tile_size, neighbor_sample_size,
+        component_sample_frequency};
+  }
+
+  static ConnectedComponentsPlan Automatic(
+      ptrdiff_t edge_tile_size = 512, uint32_t neighbor_sample_size = 2,
+      uint32_t component_sample_frequency = 1024) {
+    return {
+        kCPU, kAfforest, edge_tile_size, neighbor_sample_size,
+        component_sample_frequency};
```

Comment:
This actually doesn't match any of the plans above. To reduce the risk of this inconsistency, just call `Plan{}` and update that constructor to match the `Afforest` constructor.